### ISSUE TITLE
Warn people when their `AndroidManifest.xml` file doesn't contain the string `android:taskAffinity`

### ISSUE THREAD

**Issue by Hixie** [flutter/flutter#144484]
Subcomponent of https://github.com/flutter/flutter/issues/63559.

We should notice when an `AndroidManifest.xml` file does not include the string `android:taskAffinity` and the minSdkVersion is not sufficiently high for it to not matter, and print a warning pointing people to https://developer.android.com/privacy-and-security/risks/strandhogg, something like this:

```
Warning: this application's AndroidManifest.xml file leaves it vulnerable to the StrandHogg v1 and StrandHogg v2 Android vulnerabilities.
To mitigate the StrandHogg v1 vulnerability, set `android:taskAffinity` to the empty string in the <activity> tag. To mitigate the StrandHogg v2 vulnerability, the minSdkVersion should be set to 28 or higher.
For more details, see: https://developer.android.com/privacy-and-security/risks/strandhogg
This warning can be silenced by including the string "android:taskAffinity" somewhere in the AndroidManifest.xml file or by setting the "minSdkVersion" to a sufficiently high value.
```

**Comment by Hixie** [comment]
@andrewkolos P3 basically means "wontfix", since it's unlikely anyone else will ever get around to it. can you talk more about why this isn't something we want to do?

**Comment by christopherfujino** [comment]
re-assigning to the android team to decide the priority.

**Comment by reidbaker** [comment]
@christopherfujino does the tools team already have a way for developers to set a "dont warn me about X". 
The android team does not wish to parse the many possible manifest xml files but we do already have access to minSdkVersion and can warn but don't really want to warn every user that chooses to have a lower support version.

**Comment by christopherfujino** [comment]
> @christopherfujino does the tools team already have a way for developers to set a "dont warn me about X". The android team does not wish to parse the many possible manifest xml files but we do already have access to minSdkVersion and can warn but don't really want to warn every user that chooses to have a lower support version.

No, but maybe we should? Technically we could add a feature flag for it, but that doesn't feel right.

But a system specifically for ignoring warnings seems like it would be useful.

**Comment by reidbaker** [comment]
> > @christopherfujino does the tools team already have a way for developers to set a "dont warn me about X". The android team does not wish to parse the many possible manifest xml files but we do already have access to minSdkVersion and can warn but don't really want to warn every user that chooses to have a lower support version.
> 
> No, but maybe we should? Technically we could add a feature flag for it, but that doesn't feel right.
> 
> But a system specifically for ignoring warnings seems like it would be useful.

Filed https://github.com/flutter/flutter/issues/146130
For us to talk about this system.

**Comment by gnprice** [comment]
FWIW if I saw the warning with the wording from the original draft above:

> This warning can be silenced by including the string "android:taskAffinity" somewhere in the app's
AndroidManifest.xml file or by setting the "minSdkVersion" to a sufficiently high value.

and I wanted to silence the warning without acting on it, I'd just add a comment with that string in it. Like this:
```xml
  <!-- We don't set the android:taskAffinity attribute because [blah blah
       rationale here]. But we mention it in this comment in order to silence
       a warning from Flutter: https://github.com/flutter/flutter/issues/144484
    -->
```

After all, the warning specifically suggests "including the string" with no mention of actually putting it in as an XML attribute of an actual `<activity>` element, so it's saying that that should work.

Given that, I think it'd be reasonable to go ahead with building this warning without blocking on designing a more general "ignore this warning" system that we're happy with, i.e. #146130.

There are plenty of other potential warnings where "include this string" wouldn't work well as a rule — in particular a key aspect here is that there's that very distinctive string "android:taskAffinity", which will definitely appear if the app is addressing the issue in any normal way and on the other hand is very unlikely to appear by accident. So there's lots of other potential warnings where a system like #146130 would be needed. But I think this warning can do just fine without it.

**Comment by Hixie** [comment]
> After all, the warning specifically suggests "including the string" with no mention of actually putting it in as an XML attribute of an actual `<activity>` element, so it's saying that that should work.

For the record, that was indeed my intent.
